libdnf/dnf-sack.cpp

```diff
@@ -2147,7 +2150,9 @@ process_excludes(DnfSack *sack, GPtrArray *enabled_repos)
         }
     }
     
-    dnf_sack_add_includes(sack, &repoIncludes);
+    if (includesExists) {
+        dnf_sack_add_includes(sack, &repoIncludes);
```

Comment:
Couldn't you check that `repoIncludes` is non-empty instead of using a boolean flag?

Reply:
Using a flag is optimalization. The repoIncludes.empty() method is slow. It does iteration internaly.